### ISSUE TITLE
Running .skip will skips the current song and also the next song in the queue.

### ISSUE THREAD

**Issue by danitseitlin** [Androz2091/discord-player#1512]
**Describe the bug**
When running .skip, The current song is skipped and also the next song in the queue is skipped, for some reason it runs two times.

**To Reproduce**
1. Add 3 songs (1 to currently play and 2 in the queue)
2. Run .skip once

**Expected behavior**
Only current song is skipped.

**Comment by skdhg** [comment]
Hello @danitseitlin, I am unable to reproduce this bug, could you give more info on this?

![image](https://user-images.githubusercontent.com/46562212/212464783-1f8eea51-ac7e-4e36-846c-6d5222533e21.png)

**Comment by Drairan9** [comment]
@skdhg Bug is occurring only in 5.4.0 version
    "@discordjs/opus": "^0.8.0",
    "discord-player": "^5.4.0",
    "discord.js": "^14.7.1",
    "ffmpeg-static": "^5.1.0"

![console](https://user-images.githubusercontent.com/51862801/212563840-b27ede32-d3a4-475e-bb76-738e86f7fe9a.jpg)

Test code: https://pastebin.com/HvcVi8Af

**Comment by skdhg** [comment]
@Drairan9 Thank you I will have a look and I'm sorry I accidentally edited your comment 😅

**Comment by Siriusxm2** [comment]
For me not only .skip() skips 2 songs ahead but every song end does as well.

Another thing is songs start from the second array element when added as a playlist.

![image](https://user-images.githubusercontent.com/62480342/212770818-56f5f633-c502-4276-82c1-fb588c6f52a5.png)

![image](https://user-images.githubusercontent.com/62480342/212770878-d6b5a427-8f9e-42b3-8af3-cb0fd49c1329.png)

![image](https://user-images.githubusercontent.com/62480342/212770916-a25d0b19-dffb-4168-b0b8-eaa00f63c295.png)

Please ignore my song preferences.

**Comment by ricardo-rodrigues143** [comment]
Half of the problem seems to be in the **_createStream_** function from **_StreamDispatcher.js._** I changed the code to an earlier version and .skip is now working as expected. 
The code that I am using:

```
createStream(src, ops) {
    this.audioResource = createAudioResource(src, {
        inputType: ops?.type ?? voice_1.StreamType.Arbitrary,
        metadata: ops?.data,
        inlineVolume: true
    });

    return this.audioResource;
}
```

**Comment by febkosq8** [comment]
Can confirm, same issue. Used to work before in v5.3.2

**Comment by ikim0106** [comment]
tldr: use the code snippet below for a temporary fix. if you want a full explanation, read on!

seems to be quite the problem for many folks, I've come up with a temporary solution to this. I'm new to discord-player so I'm not sure if this problem is specific to version 5.4.0 or not, but 5.4.0 is the version I'm on.
so just to re-state the problem, it seems like skip() ends up skipping both the currently playing track and the track in a queue **if and only if there is exactly one track in the queue** (personally I have not been able to reproduce the bug in any other circumstance) so what ends up happening is that if you have one track playing and have one track in the queue, it skips **both** tracks.
as weird as it may sound, play() actually works as skip() should. it skips the current playing track and correctly moves on to the next track in the queue. but it does have a slight issue: it does not skip the currently playing track if there are no tracks in the queue.
we can combine the behavior of these two methods and come up with a solution that seems to be working as a temporary fix to the aforementioned behavior of skip()
```js
if(queue.tracks.length>0) await queue.play()
else await queue.skip()
```
the logic behind this is that play() is functionally equivalent to our idea of skip() for any situation except when the queue is empty, and that skip() is functionally equivalent to our idea of skip() for any situation except when queue has exactly one track in it.
since they are mutually exclusive, we can force the bot to run play() for any non-empty queue, and skip() for an empty queue to achieve the correct functionality.

**Comment by BombayV** [comment]
Should be fixed in the dev branch already.
